services/venice-controller/src/main/java/com/linkedin/venice/controller/ZkHelixAdminClient.java

```diff
@@ -328,4 +325,47 @@ public void close() {
   public void addInstanceTag(String clusterName, String instanceName, String tag) {
     helixAdmin.addInstanceTag(clusterName, instanceName, tag);
   }
+
+  public void setCloudConfig(VeniceControllerClusterConfig config) {
+    if (!config.isControllerCloudEnabled()) {
+      return;
+    }
+
+    String controllerCloudProvider = config.getControllerCloudProvider();
+    if (controllerCloudProvider.isEmpty()) {
+      throw new VeniceException("Controller cloud provider must be set when the controller is cloud enabled");
+    }
+
+    CloudProvider cloudProvider;
+    try {
+      cloudProvider = CloudProvider.valueOf(controllerCloudProvider.toUpperCase());
+    } catch (IllegalArgumentException e) {
+      throw new VeniceException(
+          "Invalid cloud provider: " + controllerCloudProvider + ". Must be one of: "
+              + Arrays.toString(CloudProvider.values()));
+    }
+
+    String controllerCloudId = config.getControllerCloudId();
+    List<String> controllerCloudInfoSources = config.getControllerCloudInfoSources();
+    String controllerCloudInfoProcessorName = config.getControllerCloudInfoProcessorName();
+    CloudConfig.Builder cloudConfigBuilder =
+        new CloudConfig.Builder().setCloudEnabled(true).setCloudProvider(cloudProvider).setCloudID(controllerCloudId);
+
+    if (!cloudProvider.equals(CloudProvider.AZURE)) {
```

Comment:
We should not have logic specific to AZURE, and other cloud providers. This will become a maintenance overhead. We can let Helix do the validations.

We can instead rely on the presence/absence of these configs and decide if we should set them or skip them. Helix docs say only  `CLOUD_ENABLED` and `CLOUD_PROVIDER ` are mandatory. We can apply the above checks for the rest.